src/main/java/appeng/me/storage/NetworkInventoryHandler.java

```diff
@@ -137,20 +135,20 @@ private boolean diveList( final NetworkInventoryHandler<T> networkInventoryHandl
 		return false;
 	}
 
-	private boolean testPermission( final BaseActionSource src, final SecurityPermissions permission )
+	private boolean testPermission( final IActionSource src, final SecurityPermissions permission )
 	{
-		if( src.isPlayer() )
+		if( src.player().isPresent() )
 		{
-			if( !this.security.hasPermission( ( (PlayerSource) src ).player, permission ) )
+			if( !this.security.hasPermission( src.player().get(), permission ) )
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Call "Optional#isPresent()" before accessing the value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS3655)